falcon/response_helpers.py

```diff
@@ -60,3 +61,18 @@ def format_range(value):
             str(value[0]) + '-' +
             str(value[1]) + '/' +
             str(value[2]))
+
+
+def is_ascii_encodable(s):  # pragma: no cover
+    """ check if argument encodes to ascii without error
+    """
+    if isinstance(s, six.text_type):
```

Comment:
Suggestion for a minor performance optimization: You could first check if six.PY2, and if so, simply call str(s). If it raises an error, then it was a unicode type and could not be converted to ascii, since str() does that for you, and is actually slightly faster than doing it explicitly. E.g.:

```
try:
    if six.PY2:
        str(s)
    else:
        s.encode('ascii')

except UnicodeEncodeError:
    return False

return True
```

Note that this does not check whether the input is a string type, but that isn't really what we were originally guarding against, right?